src/libsyntax/parse/parser.rs

```diff
@@ -521,7 +521,13 @@ fn dummy_arg(span: Span) -> Arg {
         span,
         id: ast::DUMMY_NODE_ID
     };
-    Arg { ty: P(ty), pat: pat, id: ast::DUMMY_NODE_ID, source: ast::ArgSource::Normal }
+    Arg {
+        ty: P(ty),
+        attrs: vec![].into(),
```

Comment:
I'm surprised that `ThinVec` doesn't implement `Default`.
So, possible improvement: implement `Default` for `ThinVec` and use it instead of `vec![].into()`.